Android compatibility

Just a single **unused** function in trantor/utils/LogStream.h, at line # 83, prevents a build in an Android app project:

```C
    void bzero()
    {
        memset(data_, 0, sizeof(data_));
    }
```
If that could be changed to "buffzero" or something like that, not only trantor, but drogon is able to be included in any Android app.

Thanks!

It happens when <strings.h> is included:

https://man7.org/linux/man-pages/man3/bzero.3.html

When trying to build in an Android environmet, the following error arrises:

In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.cpp:2:
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.h:6:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelpertemplatebox.h:7:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:83:10: error: field has incomplete type 'void'
    void bzero()
         ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:35:26: error: use of undeclared identifier 'data_'; did you mean 'data'?
    FixedBuffer() : cur_(data_)
                         ^~~~~
                         data
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:56:17: note: 'data' declared here
    const char *data() const
                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:35:21: error: member initializer 'cur_' does not name a non-static data member or base class
    FixedBuffer() : cur_(data_)
                    ^~~~~~~~~~~
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:37:19: error: use of undeclared identifier 'cookieStart'
        setCookie(cookieStart);
                  ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:42:19: error: use of undeclared identifier 'cookieEnd'
        setCookie(cookieEnd);
                  ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:49:20: error: use of undeclared identifier 'cur_'
            memcpy(cur_, buf, len);
                   ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:50:13: error: use of undeclared identifier 'cur_'
            cur_ += len;
            ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:58:16: error: use of undeclared identifier 'data_'; did you mean 'data'?
        return data_;
               ^~~~~
               data
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:56:17: note: 'data' declared here
    const char *data() const
                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:62:33: error: use of undeclared identifier 'cur_'
        return static_cast<int>(cur_ - data_);
                                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:62:40: error: use of undeclared identifier 'data_'
        return static_cast<int>(cur_ - data_);
                                       ^
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.cpp:37:
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.h:43:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelperconfigmy_program.h:5:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:83:16: error: too few arguments provided to function-like macro invocation
    void bzero()
               ^
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/strings.h:61:9: note: macro 'bzero' defined here
#define bzero(b, len) __bionic_bzero((b), (len))
        ^
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.cpp:37:
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.h:43:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelperconfigmy_program.h:5:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:83:10: error: field has incomplete type 'void'
    void bzero()
         ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:35:26: error: use of undeclared identifier 'data_'; did you mean 'data'?
    FixedBuffer() : cur_(data_)
                         ^~~~~
                         data
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:56:17: note: 'data' declared here
    const char *data() const
                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:68:16: error: use of undeclared identifier 'cur_'; did you mean 'cbrt'?
        return cur_;
               ^~~~
               cbrt
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: 'cbrt' declared here
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:35:21: error: member initializer 'cur_' does not name a non-static data member or base class
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^
    FixedBuffer() : cur_(data_)
                    ^~~~~~~~~~~
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.cpp:2:
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.h:6:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelpertemplatebox.h:7:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:68:16: error: address of overloaded function 'cbrt' does not match required type 'char'
        return cur_;
               ^~~~
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: candidate template ignored: could not match 'typename std::enable_if<std::is_integral<_A1>::value, double>::type (_A1) noexcept' against 'char'
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.cpp:2:
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.h:6:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelpertemplatebox.h:7:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:72:33: error: use of undeclared identifier 'end'
        return static_cast<int>(end() - cur_);
                                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:72:41: error: use of undeclared identifier 'cur_'; did you mean 'cbrt'?
        return static_cast<int>(end() - cur_);
                                        ^~~~
                                        cbrt
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: 'cbrt' declared here
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:37:19: error: use of undeclared identifier 'cookieStart'
        setCookie(cookieStart);
                  ^
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.cpp:2:
In file included from ../../my_program_android/MyProgramControllers/controllermodelgraph.h:6:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelpertemplatebox.h:7:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:76:9: error: use of undeclared identifier 'cur_'
        cur_ += len;
        ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:42:19: error: use of undeclared identifier 'cookieEnd'
        setCookie(cookieEnd);
                  ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:81:9: error: use of undeclared identifier 'cur_'
        cur_ = data_;
        ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:49:20: error: use of undeclared identifier 'cur_'
            memcpy(cur_, buf, len);
                   ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:50:13: error: use of undeclared identifier 'cur_'
            cur_ += len;
            ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:58:16: error: use of undeclared identifier 'data_'; did you mean 'data'?
        return data_;
               ^~~~~
               data
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:56:17: note: 'data' declared here
    const char *data() const
                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:81:16: error: use of undeclared identifier 'data_'; did you mean 'data'?
        cur_ = data_;
               ^~~~~
               data
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:56:17: note: 'data' declared here
    const char *data() const
                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:86:6: error: expected ';' after class
    }
     ^
     ;
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:62:33: error: use of undeclared identifier 'cur_'
        return static_cast<int>(cur_ - data_);
                                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:62:40: error: use of undeclared identifier 'data_'
        return static_cast<int>(cur_ - data_);
                                       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:68:16: error: use of undeclared identifier 'cur_'; did you mean 'cbrt'?
        return cur_;
               ^~~~
               cbrt
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: 'cbrt' declared here
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.cpp:37:
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.h:43:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelperconfigmy_program.h:5:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:68:16: error: address of overloaded function 'cbrt' does not match required type 'char'
        return cur_;
               ^~~~
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: candidate template ignored: could not match 'typename std::enable_if<std::is_integral<_A1>::value, double>::type (_A1) noexcept' against 'char'
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.cpp:37:
In file included from ../../my_program_android/MyProgramControllers/controllerbridgeoperations.h:43:
In file included from ../../my_program_android/MyProgramHelpers/my_programhelperconfigmy_program.h:5:
In file included from ../../my_program_android/RestServices/restservices.h:11:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/HttpController.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrObject.h:17:
In file included from ../../my_program_android/components/drogon/drogon/lib/inc/drogon/DrClassMap.h:17:
In file included from ../../my_program_android/components/drogon/drogon/trantor/trantor/utils/Logger.h:19:
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:72:33: error: use of undeclared identifier 'end'
        return static_cast<int>(end() - cur_);
                                ^
../../my_program_android/components/drogon/drogon/trantor/trantor/utils/LogStream.h:72:41: error: use of undeclared identifier 'cur_'; did you mean 'cbrt'?
        return static_cast<int>(end() - cur_);
                                        ^~~~
                                        cbrt
/home/local.user/Android/Sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/linux-x86_64/bin/../sysroot/usr/include/c++/v1/math.h:1145:1: note: 'cbrt' declared here
cbrt(_A1 __lcpp_x) _NOEXCEPT {return ::cbrt((double)__lcpp_x);}
^

... and so on

--- a/trantor/utils/LogStream.h
+++ b/trantor/utils/LogStream.h
@@ -80,7 +80,7 @@ class FixedBuffer : NonCopyable
     {
         cur_ = data_;
     }
-    void bzero()
+    void zeroBuffer()
     {
         memset(data_, 0, sizeof(data_));
     }

